libtransmission/tr-buffer.h

```diff
@@ -264,7 +265,22 @@ class StackBuffer final
 
     virtual std::pair<value_type*, size_t> reserve_space(size_t n_bytes) override
     {
-        buf_.resize(end_pos_ + n_bytes);
+        if (auto const free_at_end = buf_.size() - end_pos_; free_at_end < n_bytes)
+        {
+            if (auto const total_free = begin_pos_ + free_at_end; total_free >= n_bytes)
+            {
+                // move data so that all free space is at the end
+                auto const size = this->size();
+                std::memmove(std::data(buf_), data(), size);
```

Comment:
```suggestion
                std::copy(data(), data() + size, std::data(buf_);
```

Using `std::memmove()` is adding an extra requirement where `value_type` must be trivially copyable, not sure if this is desirable to you. Using `std::copy()` removes that requirement.

Surprisingly, it seems like using `std::copy()` might not have a performance hit. 🤔  https://stackoverflow.com/a/9980859/11390656

Reply:
Plus you don't have to worry about the overhead of the temporary buffer in `memmove()` now.

Reply:
Agreed, this is an improvement. Also this is the only use of `<cstring>` in this header so I can re-remove that.